appserver/admingui/core/src/main/java/org/glassfish/admingui/handlers/WoodstockHandler.java

```diff
@@ -193,6 +185,16 @@ public static void uploadFileToTempDir(HandlerContext handlerCtx) {
         handlerCtx.setOutputValue("uploadedTempFile", uploadTmpFile);
     }
 
+    private static File createTempFileWithOriginalFileName(String name) throws IOException {
+        final Path dir = Files.createTempDirectory(name);
+        dir.toFile().deleteOnExit();
```

Comment:
Shouldn't we delete the directory too? 
Those deleteOnExit usages will be reported by SonarQube because deleteOnExit is done just by shutdown hooks and is not something to rely on. Maybe we should target it later. Also ignoring file.delete() result can be reported by SpotBugs (which is not your code, it was already there).

- https://rules.sonarsource.com/java/type/Code%20Smell/RSPEC-2308/?search=deleteOn

Reply:
I updated the code:
* directory is also removed in the `deleteFileFromTempDir` handler
* `Files.deleteIfExists` used instead of `file.delete()`, shouldn't be reported by SpotBugs
* On IO exception, the temp files are removed if possible and an error is reported in GUI - I don't know why, but before it continued with a file, which might have been corrupted or empty